src/GoTrueAdminApi.ts

```diff
@@ -156,12 +157,18 @@ export default class GoTrueAdminApi {
    * Get a list of users.
    *
    * This function should only be called on a server. Never expose your `service_role` key in the browser.
+   * @param params An object which supports `page` and `per_page` as numbers, to alter the paginated results.
    */
-  async listUsers(): Promise<
+  async listUsers(params?: PageParams): Promise<
     { data: { users: User[] }; error: null } | { data: { users: [] }; error: AuthError }
   > {
+    let pageParams = ''
+    if (params) {
+      pageParams = params.page && params.per_page ? `?page=${params.page}&per_page=${params.per_page}` : params.page ? `?page=${params.page}` : params.per_page ? `?per_page=${params.per_page}` : ''
+    }
```

Comment:
Nit: Since the presence of params is already validated by types, you can just directly convert it to a query string, no?
```suggestion
    const pageParams = new URLSearchParams(params).toString();
```

Otherwise, you can generically clean it up beforehand:
```suggestion
    const nonFalsyParams = Object.keys(params).filter(key => params[key]).reduce((object, key) => ({ ...object, [key]: params[key] }), {})
    const pageParams = new URLSearchParams(nonFalsyParams).toString();
```

Reply:
Well that would be badass. When I try that, I get a type error though. I'm not familiar enough with typescript to know if there's a way to resolve it.

```
Type 'PageParams' is not assignable to type 'string | string[][] | Record<string, string> | URLSearchParams | undefined'.
    Type 'PageParams' is not assignable to type 'Record<string, string>'.
      Property 'page' is incompatible with index signature.
        Type 'number' is not assignable to type 'string'.
```

Reply:
Oh, interesting, didn't know `URLSearchParams` only accept strings. This should work then:
```suggestion
    const nonFalsyParams = Object.keys(params).filter(key => params[key]).reduce((object, key) => ({ ...object, [key]: params[key].toString() }), {})
    const pageParams = new URLSearchParams(nonFalsyParams).toString();
```
Also, notice that `new URLSearchParams(nonFalsyParams).toString()` wouldn't contain leading `?` symbol, so you have to add it to the url below

Reply:
Ok, so let me ask you this. In the type definition, `page` and `per_page` are numbers. If I change those types to `string`. Your one-liner code suggestion is fine - no type errors. Of course, people would then need to do:

```js
const { data, error } = await client.auth.admin.listUsers({ page: '2', per_page: '100' })
```

Which would the team prefer? It would seem weird to define them as strings, but.


If I do the two-liner, I get this type error:

```
No overload matches this call.
  Overload 1 of 2, '(o: {}): string[]', gave the following error.
    Argument of type 'PageParams | undefined' is not assignable to parameter of type '{}'.
```